apps/authorization/management/commands/repair_grants_expiration_date.py

```diff
@@ -0,0 +1,182 @@
+            print("Error: bad date time value: {}".format(created_after_date_str))
+            print(e)
+            return False
+
+        # validate date range
+        if created_after_date > created_before_date:
+            print("Error: createdbefore must >= createdafter : createdbefore = {}, createdafter = {}".format(created_before_date_str, created_after_date_str))
+            return False
+
+        # validate turn on date in date range
+        if not (created_after_date <= turn_on_date and turn_on_date <= created_before_date):
+            print("Error: turnondate {} must be in [createdafter {}: createdbefore{}]".format(turn_on_date_str, created_after_date_str, created_before_date_str))
+            return False
+
+        dryrun = options['dryrun']
+
+        if dryrun:
+            print("Info: dryrun = {}, this is a dry run, no change to database.".format(dryrun))
+            
+        apps = Application.objects.filter(name=app_name)
+
+        if not apps.exists():
+            print("Error: App {} not found.".format(app_name))
+            return False
+
+        app = apps.first()
+
+        if app is None:
+            print("Error: App '{}' not found".format(app_name))
+
+        if not ("THIRTEEN_MONTH" in app.data_access_type):
+            print("Error: This command only applies to app with THIRTEEN_MONTH data access type, '{}' data access type: {}.".format(app.name, app.data_access_type))
+            return False
+            
+        grants = DataAccessGrant.objects.filter(application=app.id).filter(created_at__range=(created_after_date, created_before_date))
+        print("process restriction: turned on date: {} and created_at in range: [{}, {}].".format(turn_on_date, created_after_date, created_before_date))
+
+        if grants.exists():
+            print("Number of grants to be checked for repair = {}".format(len(grants)))
+            processed = 0
+            before_turn_on = 0
+            after_turn_on = 0
+            for grant in grants:
+                if grant.expiration_date is not None:
+                    processed = processed + 1
+                    # repair:
+                    if grant.created_at > turn_on_date:
+                        print("+", end="")
+                        after_turn_on = after_turn_on + 1
+                        grant.expiration_date = grant.created_at.replace(tzinfo=pytz.UTC) + relativedelta(months=+13)
+                    else:
+                        print("-", end="")
+                        before_turn_on = before_turn_on + 1
+                        grant.expiration_date = turn_on_date + relativedelta(months=+13)
+                else:
+                    print("Warning: NULL expiration date encountered, command terminates.")
+                    return False
+
+                if not dryrun:
+                    print("$", end="")
+                    grant.save()
+                else:
+                    print("#", end="")
```

Comment:
All of these prints seem a little hard to interpret. Should we consider adding more detail to the output to see what the creation date, current expiration date, and expiration date after the change is? Or is that too verbose?